view/src/components/surplus.js

```diff
@@ -0,0 +1,555 @@
+import AddIcon from "@material-ui/icons/Add";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
+
+import Skeleton from "@material-ui/lab/Skeleton";
+
+const styles = (theme) => ({
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing(3),
+  },
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  submitButton: {
+    display: "block",
+    color: "white",
+    textAlign: "center",
+    position: "absolute",
+    top: 14,
+    right: 10,
+  },
+  floatingButton: {
+    position: "fixed",
+    bottom: 16,
+    right: 16,
+  },
+  form: {
+    width: "98%",
+    marginLeft: 13,
+    marginTop: theme.spacing(3),
+  },
+  toolbar: theme.mixins.toolbar,
+  root: {
+    minWidth: 470,
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  uiProgess: {
+    position: "fixed",
+    zIndex: "1000",
+    height: "32px",
+    width: "32px",
+    left: "50%",
+    top: "35%",
+  },
+  dialogStyle: {
+    maxWidth: "50%",
+  },
+  viewRoot: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+  skeleton: {
+    padding: theme.spacing(1, 1, 1, 0),
+  },
+  search: {
+    position: "relative",
+    borderRadius: theme.shape.borderRadius,
+    marginLeft: 0,
+    width: "100%",
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    width: "100%",
+  },
+  table: {
+    marginTop: "50px",
+  },
+});
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+/**
+ * Represents a Surplus component, which is a sub-page of the
+ * home page where surplus objects are visualized, created, updated,
+ * edited,and deleted.
+ */
+class Surplus extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // surplus state
+      surplusObjects: "",
+      surplusId: "",
+      produceId: "",
+      produceName: "",
+      originFarmId: "",
+      originFarmName: "",
+      originFarmLocation: "",
+      originFarmContactName: "",
+      originFarmContactPhone: "",
+      available: false,
+      cost: "",
+      totalQuantityAvailable: "",
+      packagingType: "",
+      // page state
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
```

Comment:
Maybe a comment on the context of what viewOpen is talking about / where it is used could be helpful